src/dfx-core/src/extension/manager/install.rs

```diff
@@ -104,23 +116,112 @@ impl ExtensionManager {
         crate::fs::rename(
             &temp_dir.path().join(extension_unarchived_dir_name),
             effective_extension_dir,
-        )?;
+        )
+        .map_err(|e| ExtensionError::Io(e.into()))?;
         if extension_name != effective_extension_name {
             // rename the binary
             crate::fs::rename(
                 &effective_extension_dir.join(extension_name),
                 &effective_extension_dir.join(effective_extension_name),
-            )?;
+            )
+            .map_err(|e| ExtensionError::Io(e.into()))?;
         }
         #[cfg(not(target_os = "windows"))]
         {
             let bin = effective_extension_dir.join(effective_extension_name);
-            crate::fs::set_permissions(&bin, std::fs::Permissions::from_mode(0o777))?;
+            crate::fs::set_permissions(&bin, std::fs::Permissions::from_mode(0o777))
+                .map_err(|e| ExtensionError::Io(e.into()))?;
         }
         Ok(())
     }
 }
 
+fn process_external_registry_manifest(
```

Comment:
Would you please explain the rationale for having a different scheme for storing/obtaining the manifest for external registries vs the the dfx-extensions repo?

Reply:
I see in the PR description
> dfx extension install foo --registry https://github.com/org/repo/other-compatibility.json

which seems like it was closer to what's going on in dfx-extensions.  What didn't work out with that?

Reply:
Sure, tho I'm not sure if I completely understand the question, but... the manifest from `dfinity/dfx-extensions` is of a different format than the one specified by the external manifest. As specified in [design doc](https://docs.google.com/document/d/1f6O75cPuLEO12yRxIj1FFuaVUgaPOjB9iGlC4yXm_80/edit#):
> In order for extensions stored in external repositories to be consumable by dfx, the repository maintainers must provide a URL to a JSON file with the following structure. This file is an amalgamation of the compatibility.json file and the combined ~manifest.json~ extension.json files for each individual extension, as they exist in the DFINITY extension repository.

Here are some of the differences

| | `dfinity/dfx-extensions` | external registry |
|--:|--:|--:|
| checksum  | well-known location: `https://github.com/dfinity/dfx-extensions/releases/<GIT_TAG>/downloads/<ARCHIVE_FILENAME>.sha256`  | supplied in registry file  |
| tarball download url  |  well-known location: `https://github.com/dfinity/dfx-extensions/releases/<GIT_TAG>/downloads/<ARCHIVE_FILENAME>.sha256` | supplied in registry file  |
| supported arch/platform  | determined by trying to download `https://github.com/dfinity/dfx-extensions/releases/<GIT_TAG>/downloads/<EXT>-<VER>-<ARCH>-<PLATFORM>.tar.gz` if fails then platform is not supported | determined by fetching value of `.extensions.<EXT>.<VER>.binaries.<ARCH-PLATFORM>` from registry file, if key is missing then platform is not supported |
| `extension.json` | unpacked from tarball | extracted from registry file |

To process both DFINITY extensions and external extensions using the same logic, would mean that we require extension developers to use only GitHub and mimic dir structure `dfinity/dfx-extensions`